xmtp_id/src/associations/unsigned_actions.rs

```diff
@@ -104,63 +121,67 @@ impl SignatureTextCreator for UnsignedIdentityUpdate {
-            "I authorize the following actions on XMTP:\n\n{}\n\nAuthorized at: {}",
+            "{HEADER}\n\nInbox ID: {}\nCurrent time: {}\n\n{}\n\n{FOOTER}",
+            self.inbox_id,
+            ns_date_to_iso_8601(self.client_timestamp_ns),
             all_signatures.join("\n\n"),
-            // TODO: Pretty up date
-            self.client_timestamp_ns
         )
     }
 }
 
+fn ns_date_to_iso_8601(ns_date: u64) -> String {
+    let date = DateTime::from_timestamp_nanos(ns_date as i64);
+    date.to_string()
+}
+
 #[cfg(test)]
 mod tests {
-    use crate::associations::test_utils::{rand_string, rand_u64};
+    use crate::associations::{
+        hashes::generate_inbox_id,
+        test_utils::{rand_string, rand_u64},
+    };
 
     use super::*;
 
     #[test]
     fn create_signatures() {
+        let account_address = rand_string();
+        let client_timestamp_ns = rand_u64();
         let create_inbox = UnsignedCreateInbox {
             nonce: rand_u64(),
-            account_address: rand_string(),
+            account_address: account_address.clone(),
         };
         let inbox_id = generate_inbox_id(&create_inbox.account_address, &create_inbox.nonce);
 
         let add_association = UnsignedAddAssociation {
-            inbox_id: inbox_id.clone(),
             new_member_identifier: MemberIdentifier::Address(rand_string()),
         };
 
         let revoke_association = UnsignedRevokeAssociation {
-            inbox_id: inbox_id.clone(),
             revoked_member: MemberIdentifier::Address(rand_string()),
         };
 
         let change_recovery_address = UnsignedChangeRecoveryAddress {
-            inbox_id: inbox_id.clone(),
             new_recovery_address: rand_string(),
         };
 
         let identity_update = UnsignedIdentityUpdate {
-            client_timestamp_ns: rand_u64(),
+            inbox_id: inbox_id.clone(),
+            client_timestamp_ns: client_timestamp_ns.clone(),
             actions: vec![
                 UnsignedAction::CreateInbox(create_inbox.clone()),
                 UnsignedAction::AddAssociation(add_association.clone()),
                 UnsignedAction::RevokeAssociation(revoke_association.clone()),
                 UnsignedAction::ChangeRecoveryAddress(change_recovery_address.clone()),
             ],
         };
-
+        let now_timestamp = ns_date_to_iso_8601(client_timestamp_ns);
         let signature_text = identity_update.signature_text();
-        let expected_text = format!("I authorize the following actions on XMTP:\n\nCreate Inbox: {}\n\nAdd {} to Inbox {}\n\nRemove {} from Inbox {}\n\nChange Recovery Address for Inbox {} to {}\n\nAuthorized at: {}",
-        inbox_id,
+        let expected_text = format!("{HEADER}\n\nInbox ID: {inbox_id}\nCurrent time: {now_timestamp}\n\n- Create inbox\n(Owner: {})\n\n- Grant messaging access to inbox\n(Address: {})\n\n- Revoke messaging access from app\n(Address: {})\n\n- Change inbox recovery address\n(Address: {})\n\n{FOOTER}",
```

Comment:
Could we hardcode the actual text here? You can just print the text from one test run and then paste it here. That way we can:
(1) Visually inspect the result to see if it makes sense
(2) Protect ourselves if something changes, for example the implementation of `DateTime.to_string()` changes to a different format. That would irrevocably screw up many accounts